src/main/java/reactor/netty/http/server/HttpServerBind.java

```diff
@@ -380,6 +380,10 @@ public void accept(ConnectionObserver listener, Channel channel) {
 				if (channelMetricsRecorder instanceof HttpServerMetricsRecorder) {
 					p.addAfter(NettyPipeline.HttpTrafficHandler, NettyPipeline.HttpMetricsHandler,
 							new HttpServerMetricsHandler((HttpServerMetricsRecorder) channelMetricsRecorder, uriTagValue));
+					if (channelMetricsRecorder instanceof MicrometerHttpServerMetricsRecorder) {
+						// MicrometerHttpServerMetricsRecorder does not implement metrics on protocol level
+						p.remove(handler);
```

Comment:
nitpick: reading the code it was not immediately clear `p` was the channel-level metrics and was different from the `HttpServerMetricsHandler` (applies to 3 sites)

Reply:
fixed